beets/autotag/hooks.py

```diff
@@ -371,11 +371,11 @@ def string_dist(str1: Optional[str], str2: Optional[str]) -> float:
     # Don't penalize strings that move certain words to the end. For
     # example, "the something" should be considered equal to
     # "something, the".
-    for word in SD_END_WORDS:
-        if str1.endswith(", %s" % word):
-            str1 = "{} {}".format(word, str1[: -len(word) - 2])
-        if str2.endswith(", %s" % word):
-            str2 = "{} {}".format(word, str2[: -len(word) - 2])
+    def replacer(m: re.Match[str]) -> str:
+        return f"{m.group(2)} {m.group(1)}"
+
+    str1 = re.sub(SD_END_REPLACE, replacer, str1)
+    str2 = re.sub(SD_END_REPLACE, replacer, str2)
 
```

Comment:
Neither the old option nor your version is incredibly readable. However I think I prefer the old version, because "endswith" is more readable. Could you maybe change var `m` to something more descriptive? And maybe change `replacer` to something that sounds more like a function? Like do_replace?

Reply:
That's fair.  I think a more intuitive implementation would be based on `str.rsplit()` or `str.rpartition()` using `, ` as a delimiter, and checking that the final component is one of the three words.  If that doesn't sound good, I can improve what I have here: `m` -> `match_info` and `replacer` -> `move_article_to_front`?

Reply:
I disagree with @RollingStar actually, I think your regex is clearer. Maybe I'm just more familiar/confident with regex? One thing I would note is that it might be better, if we stay with this approach, to make the regex case-insensitive.

Reply:
I've rewritten this as a much more explicit and intuitive function which uses `str.rsplit` to separate a title from an article.  @RollingStar, I hope this makes it more readable!  Also, @Serene-Arc, the inputs are lower-cased here, so case sensitivity is not an issue.